src/libs/actions/IOU.ts

```diff
@@ -7389,12 +7403,52 @@ function cancelPayment(expenseReport: OnyxEntry<OnyxTypes.Report>, chatReport: O
     );
 }
 
+/**
+ * Completes onboarding for invite link flow based on the selected payment option
+ *
+ * @param paymentSelected based on which we choose the onboarding choice and concierge message
+ */
+function completePaymentOnboarding(paymentSelected: ValueOf<typeof CONST.PAYMENT_SELECTED>) {
+    const isInviteOnboardingComplete = introSelected?.isInviteOnboardingComplete ?? false;
+
+    if (isInviteOnboardingComplete || !introSelected?.choice) {
+        return;
+    }
+
+    const session = SessionUtils.getSession();
+
+    const personalDetailsListValues = Object.values(OptionsListUtils.getPersonalDetailsForAccountIDs(session?.accountID ? [session.accountID] : [], personalDetailsList));
+    const personalDetails = personalDetailsListValues[0] ?? {};
+
+    let onboardingPurpose = introSelected.choice;
+    if (introSelected.inviteType === CONST.ONBOARDING_INVITE_TYPES.IOU && paymentSelected === CONST.IOU.PAYMENT_SELECTED.BBA) {
+        onboardingPurpose = CONST.ONBOARDING_CHOICES.MANAGE_TEAM;
+    }
+
+    if (introSelected.inviteType === CONST.ONBOARDING_INVITE_TYPES.INVOICE && paymentSelected !== CONST.IOU.PAYMENT_SELECTED.BBA) {
+        onboardingPurpose = CONST.ONBOARDING_CHOICES.CHAT_SPLIT;
+    }
+
```

Comment:
The data from BE maybe wrong in some cases. Please add two more condition 
```
if (introSelected.inviteType === CONST.ONBOARDING_INVITE_TYPES.IOU && paymentSelected !== CONST.IOU.PAYMENT_SELECTED.BBA)
```

```
if (introSelected.inviteType === CONST.ONBOARDING_INVITE_TYPES.INVOICE && paymentSelected === CONST.IOU.PAYMENT_SELECTED.BBA
```

Reply:
Why would the data from backend be incorrect? What cases do you have in mind? Those two `ifs` you've provided are already handled automatically on the backend when creating a new account from an invite link.

Reply:
Yeah I understand your point.  The BE returned the wrong data in two cases which is the reason why we already implemented two previous conditions. My suggestion will help our change to be safer and make sure that the logic works well. 

Anyway, It is not a bug. I will leave the final decision to you 😄

Reply:
I'd leave it as is - let's not meddle with the data returned from backend more than we have to for the two cases listed in the design doc